controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -354,56 +356,60 @@ public void createTransaction(CreateTxnRequest request, StreamObserver<Controlle
 
     @Override
     public void commitTransaction(TxnRequest request, StreamObserver<TxnStatus> responseObserver) {
+        final UUID txnId = ModelHelper.encode(Preconditions.checkNotNull(request.getTxnId(), "txnId"));
```

Comment:
nit: can you move preconditions inside the encode method.

Reply:
`ModelHelper.encode(...)` already has this precondition, hence removed this redundant check.